apps/src/templates/FinishDialog.jsx

```diff
@@ -0,0 +1,299 @@
+    backgroundColor: color.green,
+  },
+  continueButton: {
+    backgroundColor: color.orange,
+  },
+};
+
+export class UnconnectedFinishDialog extends Component {
+  static propTypes = {
+    isOpen: PropTypes.bool,
+    hideBackdrop: PropTypes.bool,
+
+    onContinue: PropTypes.func,
+    onReplay: PropTypes.func,
+
+    isPerfect: PropTypes.bool,
+    blocksUsed: PropTypes.number,
+    blockLimit: PropTypes.number,
+    achievements: PropTypes.arrayOf(PropTypes.shape({
+      isAchieved: PropTypes.bool,
+      iconUrl: PropTypes.string,
+      message: PropTypes.string,
+    })),
+    showFunometer: PropTypes.bool,
+    canShare: PropTypes.bool,
+  };
+
+  getBubble() {
+    return (
+      <div style={styles.bubbleContainer}>
+        <div
+          style={{
+            ...styles.bubble,
+            backgroundColor:
+              this.props.isPerfect ? color.level_perfect : color.level_passed,
+          }}
+        />
+      </div>
+    );
+  }
+
+  getBlockCountDescription() {
+    if (this.props.blocksUsed < this.props.blockLimit) {
+      return msg.betterThanPerfectDescription();
+    }
+    if (this.props.blocksUsed === this.props.blockLimit) {
+      return msg.perfectDescription();
+    }
+    if (this.props.blocksUsed > this.props.blockLimit) {
+      return msg.tooManyBlocksDescription();
+    }
+  }
+
+  getBlockCounter() {
+    if (this.props.blockLimit === undefined) {
+      return null;
+    }
+
+    const tooManyBlocks = this.props.blocksUsed > this.props.blockLimit;
+    return (
+      <div style={styles.blockCountWrapper}>
+        <span style={styles.blockCountLabel}>
+          {msg.numBlocksUsedLabel()}:
+        </span>
+        <span
+          style={tooManyBlocks ?
+            styles.blockCountPass : styles.blockCountPerfect}
+        >
+          <span style={styles.blockCount}>
+            {(this.props.blocksUsed || 0).toString()}
+            {this.props.blockLimit && ('/' + this.props.blockLimit.toString())}
+          </span>
+          <span style={styles.blockCountDescriptor}>
+            {this.getBlockCountDescription()}
+          </span>
+        </span>
+      </div>
+    );
+  }
+
+  getAchievements() {
+    if (!this.props.achievements || this.props.achievements.length === 0) {
+      return null;
+    }
+
+    return (
+      <div style={styles.achievements}>
+        {this.props.achievements.map((achievement, index) => (
+          <div
+            key={index}
+            style={{
+              ...styles.achievementRow,
+              ...(achievement.isAchieved ? {} : {color: color.lighter_gray}),
+            }}
+          >
+            <img href={achievement.iconUrl} style={styles.achievementIcon} />
+            {achievement.message}
+          </div>
+        ))}
+      </div>
+    );
+  }
+
+  getFunometer() {
+    if (!this.props.showFunometer) {
+      return null;
+    }
+
+    return (
+      <div style={styles.funometer}>
+        <PuzzleRatingButtons
+          useLegacyStyles
```

Comment:
I think we can get rid of the `newStyles` now that the Gamification experiment no longer uses them?  (Follow up PR..)